### PR TITLE
Fix @PreHandler `methods` default value

### PR DESCRIPTION
**Pull request by shiliyu**

### CODE REVIEW

File: core/src/test/java/io/forestframework/core/http/routing/PreHandlerIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+class PreHandlerTestApp {
+
+    @PreHandler(value = "/prehandler/*")
+    public void preHandle(HttpServerRequest request, HttpServerResponse response) {
+        response.write(request.path(), "UTF-8");
+    }
+
+    @PreHandler("/prehandler/error/*")
+    public boolean preHandleException(HttpServerRequest request, HttpServerResponse response) {
+        response.setStatusCode(503).write("Service Unavailable for " + request.path());
```
Comment (blindpirate) [inline core/src/test/java/io/forestframework/core/http/routing/PreHandlerIntegrationTest.java:37]: Where did you assert status code 503?

File: core/src/test/java/io/forestframework/core/http/routing/PreHandlerIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    @Route(path = "/prehandler/*", type = RoutingType.HANDLER)
+    public void handle(HttpServerResponse response) {
+        response.write(" is handled", "UTF-8");
+    }
+
+    @Route(path = "/prehandler/error/*", type = RoutingType.HANDLER)
+    public void handleException(HttpServerResponse response) {
+        response.write(" should not be here", "UTF-8");
```
Comment (blindpirate) [inline core/src/test/java/io/forestframework/core/http/routing/PreHandlerIntegrationTest.java:48]: Where did you assert "response NOT contains `should not be here`"?

File: core/src/test/java/io/forestframework/core/http/routing/PreHandlerIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    }
+}
+
+@DisplayName("When pre-handler methods default")
+@ExtendWith(ForestExtension.class)
+@ForestTest(appClass = PreHandlerTestApp.class)
```
Comment (blindpirate) [inline core/src/test/java/io/forestframework/core/http/routing/PreHandlerIntegrationTest.java:54]:
```suggestion
@ForestTest(appClass = PreHandlerTestApp.class)
@DisableAutoScan
```

File: core/core.gradle.kts [hunk 4]
```diff
@@ -1,6 +1,7 @@
 val libs: (String) -> String by rootProject.ext
 
 dependencies {
+    api("org.apache.httpcomponents:httpclient:4.5.12")
```
Comment (blindpirate) [inline core/core.gradle.kts]: 1. It should be `testImplementation`.
2. Please move this notation to `dependencies.gradle.kts`, then use `testImplementation(libs("httpclient"))` here, just like other dependencies.

**Review by blindpirate — CHANGES_REQUESTED** [review]

File: core/core.gradle.kts (hunk 4 above)
Comment (shiliyu) [reply core/core.gradle.kts]: 'kay, will do.

File: core/src/test/java/io/forestframework/core/http/routing/PreHandlerIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+    @ParameterizedTest(name = "should handle \"{0}\" method")
+    @CsvSource({
+            "GET, /prehandler/get, /prehandler/get is handled",
+            "POST, /prehandler/post, /prehandler/post is handled",
+            "PATCH, /prehandler/patch, /prehandler/patch is handled",
+            "DELETE, /prehandler/delete, /prehandler/delete is handled"
+    })
+    void should_handle_all_methods_when_methods_attribute_is_default_value_in_preHandler(String method, String path, String expected) throws IOException {
+        Assertions.assertEquals(expected, sendRequestandGetResult(method, path));
```
Comment (blindpirate) [inline core/src/test/java/io/forestframework/core/http/routing/PreHandlerIntegrationTest.java]: Can you assert 200 here?

File: core/src/test/java/io/forestframework/core/http/routing/PreHandlerIntegrationTest.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+            "PATCH, /prehandler/error/patch, Service Unavailable for /prehandler/error/patch",
+            "DELETE, /prehandler/error/delete, Service Unavailable for /prehandler/error/delete"
+    })
+    void should_handle_all_methods_when_methods_attribute_is_default_value_in_preHandler_and_return_when_error_occurs(String method, String path, String expected) throws IOException {
+        CloseableHttpResponse response = sendRequestGetResponse(method, path);
+        int statusCode = response.getStatusLine().getStatusCode();
+
+        Assertions.assertEquals(503, statusCode);
+        Assertions.assertEquals(expected, sendRequestandGetResult(method, path));
+        Assertions.assertFalse(sendRequestandGetResult(method, path).contains("should not be here"));
```
Comment (blindpirate) [inline core/src/test/java/io/forestframework/core/http/routing/PreHandlerIntegrationTest.java]: Why don't you request once, assert twice? What you're doing right now is wasting time.

File: core/src/test/java/io/forestframework/core/http/routing/PreHandlerIntegrationTest.java [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+    void setUp() {
+        client = HttpClients.createDefault();
+    }
+
+    @ParameterizedTest(name = "should handle \"{0}\" method")
+    @CsvSource({
+            "GET, /prehandler/get, /prehandler/get is handled",
+            "POST, /prehandler/post, /prehandler/post is handled",
+            "PATCH, /prehandler/patch, /prehandler/patch is handled",
+            "DELETE, /prehandler/delete, /prehandler/delete is handled"
```
Comment (blindpirate) [inline core/src/test/java/io/forestframework/core/http/routing/PreHandlerIntegrationTest.java]: Nit: you can only use one parameter for this method:

```
void should_handle_all_methods_when_methods_attribute_is_default_value_in_preHandler(String method) {
     String path = "/prehandler/" + method.toLowerCase();
     String expected = "/prehandler/" + method.toLowerCase() + " is handled";
}
```

File: core/src/test/java/io/forestframework/core/http/routing/PreHandlerIntegrationTest.java [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+public class PreHandlerIntegrationTest {
+
+    @Inject
+    @Config("forest.http.port")
+    Integer port;
+
+    CloseableHttpClient client;
+
+    @BeforeEach
+    void setUp() {
```
Comment (blindpirate) [inline core/src/test/java/io/forestframework/core/http/routing/PreHandlerIntegrationTest.java:65]: Please make all test method (including `setUp()`) public.

**Review by blindpirate — CHANGES_REQUESTED** [review]

**Comment by blindpirate** [comment]
Can you also fix the checkstyle issues?

File: core/src/test/java/io/forestframework/core/http/routing/PreHandlerIntegrationTest.java (hunk 7 above)
Comment (shiliyu) [reply core/src/test/java/io/forestframework/core/http/routing/PreHandlerIntegrationTest.java]: oh yes, this is much better.

File: core/src/test/java/io/forestframework/core/http/routing/PreHandlerIntegrationTest.java (hunk 6 above)
Comment (shiliyu) [reply core/src/test/java/io/forestframework/core/http/routing/PreHandlerIntegrationTest.java]: 'kay, will do.

File: core/src/test/java/io/forestframework/core/http/routing/PreHandlerIntegrationTest.java (hunk 5 above)
Comment (shiliyu) [reply core/src/test/java/io/forestframework/core/http/routing/PreHandlerIntegrationTest.java]: Sure.

File: core/src/main/java/io/forestframework/ext/core/AutoRoutingScanExtension.java [hunk 9]
```diff
@@ -55,7 +55,8 @@ public void afterInjector(Injector injector) {
         RoutingManager routings = injector.getInstance(RoutingManager.class);
 
         // @formatter:off
-        List<Class<?>> componentClasses = injector.getInstance(Key.get(new TypeLiteral<List<Class<?>>>() { }, ComponentClasses.class));
+        List<Class<?>> componentClasses = injector.getInstance(Key.get(new TypeLiteral<List<Class<?>>>() {
```
Comment (blindpirate) [inline core/src/main/java/io/forestframework/ext/core/AutoRoutingScanExtension.java]: > [ant:checkstyle] [ERROR] /home/runner/work/forest-petclinic/forest-petclinic/core/src/main/java/io/forestframework/ext/core/AutoRoutingScanExtension.java:143:55: '{' is not preceded with whitespace. [WhitespaceAround]

The error says line 143, Why do you change this line?

Comment (shiliyu) [reply core/src/main/java/io/forestframework/ext/core/AutoRoutingScanExtension.java]: i swear i didnt touch this line, maybe i randomly touched the keyboard. SORRY, will double check.

**Review by blindpirate — APPROVED** [review]

**Pull request merged**